src/randomenv.cpp

```diff
@@ -357,10 +357,21 @@ void RandAddStaticEnv(CSHA512& hasher)
     hasher << &hasher << &RandAddStaticEnv << &malloc << &errno << &environ;
 
     // Hostname
+#ifdef WIN32
+    // Not using gethostname from Windows Sockets because
+    // the use of the Winsock DLL might be not initiated.
```

Comment:
>     // the use of the Winsock DLL might be not initiated.

Why not?

Reply:
When using `libbitcoinkernel`, it depends on its user, no?

Reply:
I don't know, given the kernel and it's API/runtime requirements doesn't exist yet. It's also less clear, because there is nothing else in this codebase documenting this requirement, or preventing winsock usage (or any other DLL that might require manual initialization) from inadvertantly ending up back in the kernel.

Reply:
> ... there is nothing else in this codebase ... preventing winsock usage (or any other DLL that might require manual initialization) from inadvertantly ending up back in the kernel.

Well, the upcoming CMake-based build system will likely [complain](https://github.com/hebasto/bitcoin/pull/134) if Winsock dependency is about to be reintroduced.

Reply:
> CMake-based build system will likely complain

Complain about what? Putting aside all the other points here, why are we changing code in our RNG, supposedly based on some thing that CMake may or may not do, that you didn't mention at all in the PR description?

Reply:
Perhaps, I didn't understand your point correctly. Sorry about that.

This PR goal is to remove winsock dependency from libbitcoinkernel. The current build system does not prevent from reintroducing this dependency inadvertantly. But CMake does.

Reply:
> it comes somewhat from vauge / useless comments

They are dropped now.

Reply:
>  then a test or similar should be added that actually enforces that

Could be something as simple as doing a symbol check for `bitcoin-chainstate` and ensuring `WS2_32` is not in its import table. I guess we should be doing this anyway at some point, but not sure where I would put such a check for non-release, experimental binaries.

Reply:
> I guess we should be doing this anyway at some point,

I will try follow up here with something more globally applicable.